src/exec/use_pty/backchannel.rs

```diff
@@ -29,8 +29,6 @@ pub(super) struct BackchannelPair {
 impl BackchannelPair {
     pub(super) fn new() -> io::Result<Self> {
         let (sock1, sock2) = UnixStream::pair()?;
-        sock1.set_nonblocking(true)?;
```

Comment:
maybe we could keep these in the debug builds (`#[cfg(debug_assertions)]` and add a `debug_assert!` that checks that reading / writing to the sockets never yields a `ErrorKind::WouldBlock`